### PR TITLE
Some additional rename suggestions and converting instanceid to collectionid

### PR DESCRIPTION
**Pull request by ashanhol**
Some additional rename suggestions and converting instanceid to collectionid 

-----------
Reminder:

PRs merged into main will not automatically generate a PR in https://github.com/broadinstitute/terra-helmfile to update the WDS image deployed to kubernetes - this action will need to be triggered manually by running the following github action: https://github.com/DataBiosphere/terra-workspace-data-service/actions/workflows/tag.yml. Dont forget to provide a Jira Id when triggering the manual action, if no Jira ID is provided the action will not fully succeed. 

After you manually trigger the github action (and it completes with no errors), you must go to [the terra-helmfile](https://github.com/broadinstitute/terra-helmfile) repo and verify that this generated a PR that merged successfully.

The terra-helmfile PR merge will then generate a PR in [leonardo](https://github.com/DataBiosphere/leonardo).  This will automerge if all tests pass, but if jenkins tests fail it will not; be sure to watch it to ensure it merges. To trigger jenkins retest simply comment on PR with "jenkins retest".

### CODE REVIEW

**Review by davidangb — CHANGES_REQUESTED** [review]
we can't rename anything in `apis-v1.yaml`. Doing so will cascade changes to our generated clients - you can see in this PR that some of the generated code was updated. This will break anyone using those clients.

To rename anything in `apis-v1.yaml` we need to create new v2 or v3 (etc) APIs, deprecate the v1 APIs, and fully remove the v1 APIs after some time.

File: service/src/main/resources/static/swagger/apis-v1.yaml [hunk 1]
```diff
@@ -189,15 +189,15 @@ components:
         - status
         - created
         - updated
-        - instanceId
+        - collectionId
       properties:
```
Comment (calypsomatic) [inline service/src/main/resources/static/swagger/apis-v1.yaml:193]: Ooh yeah I'm trying not to change anything in the APIs yet, even though it's obnoxious.

File: service/src/test/java/org/databiosphere/workspacedataservice/controller/JobControllerMockMvcTest.java [hunk 2]
```diff
@@ -61,7 +61,7 @@ void smokeTestJobsInInstanceV1() throws Exception {
     // return a test job from the mocked JobDao
     UUID jobId1 = UUID.randomUUID();
     UUID jobId2 = UUID.randomUUID();
-    InstanceId instanceId = new InstanceId(UUID.randomUUID());
+    CollectionId collectionId = new CollectionId(UUID.randomUUID());
```
Comment (calypsomatic) [inline service/src/test/java/org/databiosphere/workspacedataservice/controller/JobControllerMockMvcTest.java:64]: Also deliberately leaving out the controller layer - since I associate that with the APIs I was going to leave this layer for a separate PR

Comment (ashanhol) [reply service/src/test/java/org/databiosphere/workspacedataservice/controller/JobControllerMockMvcTest.java:64]: This isn't actually renaming anything in the controller, just converting any use of InstanceId to CollectionId. But that rename also had to happen for the generated job model (which is why i modified the v1 api). I can go through and see if I can walk it back but if they're attached we might want to wait to rename the InstanceId.java
